docs/developers/extending.rst

```diff
@@ -0,0 +1,157 @@
+
+#. Adding new plugins directly in Open Forms. These are extensions that serve the greater community and can be
+   contributed as a pull request to Open Forms (more information can be found in the `CONTRIBUTING.md`_).
+
+#. Adding 'external' plugins. These are extensions that are extremely specific or can't be open-sourced.
+   Instructions on how to implement such 'external plugins' are described below.
+
+.. _CONTRIBUTING.md: https://github.com/open-formulieren/open-forms/blob/master/CONTRIBUTING.md
+
+Creating an external plugin
+===========================
+
+The `demo extension <https://github.com/open-formulieren/demo-extension>`_ is an example of an extension plugin.
+
+An extension must be implement as a Django package, which is a stricter form of a
+regular Python package. Extension code is allowed to import from Open Forms' public API, as
+documented :ref:`here<developers_backend_core_index>`.
+
+
+Steps to implement an extension plugin
+--------------------------------------
+
+#. You can use the `default-app`_ as a template for an empty django package. Follow the instructions in the ``README.rst`` file to start a new Django project.
+
+#. Implement the plugin in the ``<project_name>`` directory. This will require at least updating/adding the files below. Refer to the :ref:`plugin section<plugins_index>` for further details.
+
+   * ``apps.py``. See for example `apps.py`_.
+
+   * ``plugin.py``. See for example `plugin.py`_. This is where the functionality of the plugin will be implemented.
+
+.. _plugin.py: https://github.com/open-formulieren/demo-extension/blob/main/demo_extension/plugin.py
+.. _apps.py: https://github.com/open-formulieren/demo-extension/blob/main/demo_extension/apps.py
+.. _default-app: https://github.com/maykinmedia/default-app
+
+Configurable options
+^^^^^^^^^^^^^^^^^^^^
+
+Extensions cannot require modifications to the Django settings of Open Forms. Any run-time configuration option can
+be specified as:
+
+#. Deployment time environment variables. See `plugin.py:35 <https://github.com/open-formulieren/demo-extension/blob/main/demo_extension/plugin.py#L35>`_.
+
+#. Dynamic options using ``django-solo``, with the advantage that configuration can be
+   modified at runtime through the Admin interface. See `models.py`_ and `admin.py`_ in the demo plugin.
+   If you add a solo model, you will have to generate migrations.
+   See the :ref:`section below<test_extension_plugin_locally>` for details.
+
+#. For certain types of plugins, like for registration backends, it is also possible to add form-specific options.
+   These can be specified with a serializer. See `config.py`_ for details.
+
+If any initialization or module loading is needed, this should be done as part of the
+extension's ``extension.apps.ExtensionConfig.ready`` hook, which is called when Django
+bootstraps. This is the mechanism by which Open Forms plugins are registered
+(refer to the :ref:`plugin section<plugins_index>` for further details).
```

Comment:
this looks copied a bit too-verbatim from the issue ;)

Reply:
:sweat_smile: